linalg/solvers.cpp

```diff
@@ -413,9 +410,16 @@ void CGSolver::Mult(const Vector &b, Vector &x) const
       MFEM_ASSERT(IsFinite(den), "den = " << den);
       if (den <= 0.0)
       {
-         if (print_level >= 0 && Dot(d, d) > 0.0)
+         if (Dot(d, d) > 0.0 && print_level >= 0)
```

Comment:
To avoid the extra computation when `print_level` is set to `0`
```suggestion
      if (print_level >= 0 && Dot(d, d) > 0.0)
```

Reply:
Same reasoning as above applies here.